.github/workflows/build.yml

```diff
@@ -32,6 +34,8 @@ jobs:
           REPO_NAME=$(echo "${{ github.repository }}" | tr '[:upper:]' '[:lower:]')
           echo "image-tag=${BRANCH_NAME}" >> $GITHUB_OUTPUT
           echo "repo-name=${REPO_NAME}" >> $GITHUB_OUTPUT
+          echo "dep-matrix=$(jq -r '.dep' -c .github/workflows/matrix.json)" >> $GITHUB_OUTPUT
+          echo "src-matrix=$(jq -r '.src' -c .github/workflows/matrix.json)" >> $GITHUB_OUTPUT
```

Comment:
I kinda hate using jq in the build.yml like this; but I guess it's ok

Reply:
The alternative would be having to maintain two JSON files (they would have to be JSON if we want to share params between them without resorting to duplication), which wouldn't be the worst thing in the world but imo since `src` will be reading from `dep` for values and they are configuring the same actions run, just different jobs, consolidation would be better.

Plus we'd be using `jq` later on regardless to read through the matrix definition later on to find `dep_opts` and `host` from `depends_name`.

Reply:
these 2 JSON seems doens't have any dependency on each other, maybe better to have 2 files?

Is there any case, when you need both of them at once?

Reply:
Because they the second section (`src`) refers to the first section (`dep`) later on and they ultimately represent two halves of one workflow.

https://github.com/dashpay/dash/blob/3cc8c70fcc6d14eeddb3600f815a8e8643d47ee7/.github/workflows/build.yml#L126-L136

Examples of these are explained in the PR description.